src/App.tsx

```diff
@@ -1,14 +1,64 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [filter, setFilter] = useState('');
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  let visibleTodos = todos;
+
+  if (searchQuery) {
+    visibleTodos = visibleTodos.filter(todo => todo.title.toLowerCase().includes(searchQuery.toLowerCase()));
+  }
+
+  if (filter) {
+    switch (filter) {
+      case 'all': {
+        break;
+      }
+
+      case 'active': {
+        visibleTodos = visibleTodos.filter(todo => todo.completed === false);
+        break;
+      }
+
+      case 'completed': {
+        visibleTodos = visibleTodos.filter(todo => todo.completed === true);
+        break;
+      }
+
+      default:
+        break;
+    }
+  }
```

Comment:
I think, in this task, you can simplify code without using switch but lets see what mentors will say)